Controllers/conversationController.js

```diff
@@ -79,25 +79,48 @@ exports.index = CatchAsyncError(async (req, res, next) => {
 })
 
 exports.show = CatchAsyncError(async (req, res, next) => {
-  const conversation = await Conversation.findAll({
-    raw: true,
-    nest: true,
-    attributes: ['id'],
-    where: {
-      id: req.params.id
-    },
+  const messages = await Conversation.findOne({
+    attributes: ['id', 'createdAt', 'updatedAt'],
     limit: req.query.messagecount,
     include: [
       {
         model: Message,
-        attributes: ['body', 'userId'],
+        attributes: ['body', 'userId', 'createdAt', 'updatedAt'],
         where: {
           conversationId: req.params.id
         }
       }
     ]
   })
-  if (conversation.length > 0) {
+  const users = await Conversation.findOne({
+    where: {
+      id: req.params.id
+    },
+    include: [
+      {
+        model: User,
+        attributes: ['username', 'photoURL', 'id'],
+        through: {
+          attributes: []
+        }
+      }
+    ]
+  })
```

Comment:
You can get the conversation and the users in a single query. Since sequelize does not support limiting included models you should make a separate query for the messages, so that pagination will work.